src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -1578,22 +1591,41 @@ export class PostgresQueryRunner
                     true,
                 )
 
+                // when only adding enum value(s), we can use alternative syntax
+                if (useAddValueForUp) {
+                    for (const item of extraItems) {
+                        const escapedValue = item.replace("'", "''")
+
+                        upQueries.push(
+                            new Query(
+                                `ALTER TYPE ${oldEnumName} ADD VALUE '${escapedValue}'`,
+                            ),
+                        )
+                    }
+                }
+
                 // rename old ENUM
-                upQueries.push(
-                    new Query(
-                        `ALTER TYPE ${oldEnumName} RENAME TO ${oldEnumNameWithoutSchema_old}`,
-                    ),
-                )
+                if (!useAddValueForUp) {
+                    upQueries.push(
+                        new Query(
+                            `ALTER TYPE ${oldEnumName} RENAME TO ${oldEnumNameWithoutSchema_old}`,
+                        ),
+                    )
+                }
+
                 downQueries.push(
                     new Query(
                         `ALTER TYPE ${oldEnumNameWithSchema_old} RENAME TO ${oldEnumNameWithoutSchema}`,
                     ),
                 )
```

Comment:
When `useAddValueForUp` is true, the `oldEnumNameWithSchema_old` won't have been created, so this down query isn't going to work?

Interspersing `if (!useAddValueForUp)` conditionals throughout this code is making the outcome hard to reason about. Instead, I think it would be preferable to separate these code paths at a higher level. (This may mean a few bits of duplicated code where the behavior is the same in both paths, but that's acceptable.)